### PR TITLE
add grpcio

### PR DESCRIPTION
**Pull request by ocefpaf**
See https://pypi.org/project/grpcio/

### CODE REVIEW

**Comment by ocefpaf** [comment]
@conda-forge/core this recipe is a `tensoflow` dependency that is not present on `conda-forge` and, because of the channel mixing issues we are experiencing lately, installing `conda-forge`'s `tensorflow` will pull many packages from `defaults` and create a possible inconsistent env depending on what else you have there.

I'm completely stuck here and in https://github.com/conda-forge/staged-recipes/pull/6523 (a possible alternative), so any help is very welcomed.

PS: we'll have a workshop next week where I was planning to wear my conda-forge t-shirt and hand out anvil stickers. I guess I won't do that until we can provide a stable :grimacing:

**Comment by scopatz** [comment]
It looks like it is trying to execute using `gcc` rather than `g++`.  Can you set `CXX=g++` or similar?

**Comment by ocefpaf** [comment]
> It looks like it is trying to execute using `gcc` rather than `g++`.  Can you set `CXX=g++` or similar?

https://github.com/conda-forge/toolchain-feedstock/blob/master/recipe/activate.sh#L23

Maybe similar b/c `CXX=g++`is done in the `toolchain` already :-(

PS: I can build it locally without the `toolchain` but got tons of erros when trying to figure out which flag was the culprit. Maybe I am missing something obvious here.

**Comment by jjhelmus** [comment]
You may need to add `-std=c++11` to the GRPC_PYTHON_CFLAGS environmental variable.

**Comment by isuruf** [comment]
@ocefpaf, can you send the log that succeeds without toolchain?

File: recipes/grpcio/set_109_deployment_target.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/setup.py b/setup.py
+index 67c36a9..7b1b709 100644
+--- a/setup.py
++++ b/setup.py
+@@ -193,7 +193,7 @@ if 'darwin' in sys.platform and PY3:
+   mac_target = sysconfig.get_config_var('MACOSX_DEPLOYMENT_TARGET')
+   if mac_target and (pkg_resources.parse_version(mac_target) <
+                      pkg_resources.parse_version('10.7.0')):
+-    os.environ['MACOSX_DEPLOYMENT_TARGET'] = '10.7'
++    os.environ['MACOSX_DEPLOYMENT_TARGET'] = '10.9'
```
Comment (isuruf) [inline recipes/grpcio/set_109_deployment_target.patch]: Is this necessary? `MACOSX_DEPLOYMENT_TARGET` is 10.9 and the if condition is false

**Comment by isuruf** [comment]
Tests pass now on all 3 platforms. I removed building vendored copies of boringssl, zlib and c-ares and used openssl, zlib, c-ares from conda-forge.

**Comment by ocefpaf** [comment]
> Tests pass now on all 3 platforms. I removed building vendored copies of boringssl, zlib and c-ares and used openssl, zlib, c-ares from conda-forge.

Wow! Awesome!! Thanks a lot @isuruf!!!

**Comment by ocefpaf** [comment]
> @ocefpaf, can you send the log that succeeds without toolchain?

Too late but here is goes:

https://gist.github.com/ocefpaf/e744cddc942e257a5fb331d97c7b550b

**Comment by jakirkham** [comment]
It seems like the `clang` symlinks are causing some issue with conversion.

```python
Traceback (most recent call last):
  File "/home/travis/miniconda/lib/python3.6/site-packages/conda_smithy/cli.py", line 33, in generate_feedstock_content
    configure_feedstock.copytree(source_recipe_dir, target_recipe_dir)
  File "/home/travis/miniconda/lib/python3.6/site-packages/conda_smithy/configure_feedstock.py", line 62, in copytree
    copy_file(s, d)
  File "/home/travis/miniconda/lib/python3.6/site-packages/conda_smithy/feedstock_io.py", line 85, in copy_file
    with io.open(src, "r", encoding="utf-8") as fh_src:
FileNotFoundError: [Errno 2] No such file or directory: '/home/travis/build/conda-forge/staged-recipes/recipes/grpcio/clang'
```

ref: https://travis-ci.org/conda-forge/staged-recipes/builds/418300333

**Comment by ocefpaf** [comment]
@isuruf I guess these symlinks are preventing the recipe from becoming a feedstock:

https://github.com/conda-forge/staged-recipes/blob/master/recipes/grpcio/clang

**Pull request merged**